src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -64,6 +64,13 @@ public function setFactory($factory)
             $factory = explode('::', $factory, 2);
         }
 
+        if (is_array($factory) && !isset($factory[0])) {
```

Comment:
We shouldn't support empty arrays, but there wasn't such checks before.
I can explicitly check for `null`, without the isset. then it'll fail badly in case of an empty array.

I think the fact the array should contain 2 elements is almost part of the signature (through phpdoc), hence there is not any check about it. If we add them, it should be considered as a bug fix I guess?